assets/wizards/engagement/views/reader-activation/index.js

```diff
@@ -52,7 +53,18 @@ export default withWizardScreen( () => {
 			.catch( setError )
 			.finally( () => setInFlight( false ) );
 	};
+	const fetchPluginStatus = () => {
+		setError( false );
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack/v1/wizard/newspack-engagement-wizard/reader-activation-plugins',
```

Comment:
Why another request – the plugin status could be a part of the response from `GET /newspack-engagement-wizard/reader-activation` request.

Reply:
It was that way originally, but doing so tied the plugins check to config that's fetched from and saved to the options table. So what was happening was that the check only occurred once and then the success state was saved to `wp_options` the first time all required plugins were available.

I tried separating the check from the other options in the same request, but it resulted in a lot of unnecessary re-renders of the component, so I thought separating the plugin status check from the options table settings would ensure that we were always getting the current state of the plugins without causing a re-render if not needed.

Reply:
> I tried separating the check from the other options in the same request, but it resulted in a lot of unnecessary re-renders of the component

I don't understand. Have you tried it like so – https://github.com/Automattic/newspack-plugin/commit/6745d23d2eefc8fe52f002b4e928fda5c1594201 ?

Reply:
No, no I didn't. 😆 I was trying to fit the plugin status into the existing `config` object, but your solution is much better.

Reply:
cca1393 implements the single fetch and gets rid of the extra REST route. No extra re-renders!